Whiskey/Tasks/Invoke-WhiskeyDotNetBuild.ps1

```diff
@@ -0,0 +1,139 @@
+
+    # Examples
+
+    ## Example 1
+
+        BuildTasks:
+        - DotNetBuild:
+            Path:
+            - DotNetCoreSolution.sln
+
+    Demonstrates building the DotNetCoreSolution.sln file with the `dotnet build` command.
+
+    ## Example 2
+
+        BuildTasks:
+        - DotNetBuild:
+            Path:
+            - DotNetCoreSolution.sln
+            Verbosity: normal
+            OutputDirectory: bin
+
+    Demonstrates build a solution file with normal verbosity and compiling the assemblies to the '$(WHISKEY_BUILD_ROOT)\bin' directory.
+
+    ## Example 3
+
+        BuildTasks:
+        - DotNetBuild:
+            Path:
+            - src\DotNetStandardLibrary.csproj
+            - src\DotNetCoreApp.csproj
+            Argument:
+            - --no-dependencies
+
+    Demonstrates building multiple .NET Core csproj files with an additional argument, `--no-dependencies`, passed to the `dotnet build` command.
+
+    ## Example 4
+
+        BuildTasks:
+        - DotNetBuild:
+            Path:
+            - DotNetCoreSolution.sln
+            SDKVersion: 2.*
+
+    Demonstrates building a .NET Core solution with the latest `2.*` version of the .NET Core SDK.
+    #>
+    [CmdletBinding()]
+    [Whiskey.Task("DotNetBuild")]
+    [Whiskey.RequiresTool('Dotnet','DotnetPath',VersionParameterName='SDKVersion')]
+    param(
+        [Parameter(Mandatory=$true)]
+        [object]
+        $TaskContext,
+
+        [Parameter(Mandatory=$true)]
+        [hashtable]
+        $TaskParameter
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    Set-Item -Path 'env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE' -Value $true
+    Set-Item -Path 'env:DOTNET_CLI_TELEMETRY_OPTOUT' -Value $true
+
+    $dotnetExe = $TaskParameter['DotnetPath']
+
+    $projectPaths = ''
+    if ($TaskParameter['Path'])
+    {
+        $projectPaths = $TaskParameter['Path'] | Resolve-WhiskeyTaskPath -TaskContext $TaskContext -PropertyName 'Path'
+    }
+
+    $verbosity = $TaskParameter['Verbosity']
+    if (-not $verbosity -and $TaskContext.ByBuildServer)
+    {
+        $verbosity = 'detailed'
+    }
+
+    $outputDirectory = $TaskParameter['OutputDirectory']
+    if ($outputDirectory)
+    {
+        $outputDirectory = $outputDirectory | Resolve-WhiskeyTaskPath -TaskContext $TaskContext -PropertyName 'OutputDirectory' -Force
+    }
+
+    $dotnetArgs = & {
+        '--configuration={0}' -f (Get-WhiskeyMSBuildConfiguration -Context $TaskContext)
+        '-p:Version={0}'      -f $TaskContext.Version.SemVer1.ToString()
+
+        if ($verbosity)
+        {
+            '--verbosity={0}' -f $verbosity
+        }
+
+        if ($outputDirectory)
+        {
+            '--output={0}' -f $outputDirectory
+        }
+
+        $TaskParameter['Argument']
+    }
+
+    Write-WhiskeyVerbose -Context $TaskContext -Message     (' dotnet.exe command path:    {0}' -f $dotnetExe)

```

Comment:
What were you wanting the verbose messages to look like this case? Following this format I get the following, which looks a little confusing:
```
VERBOSE: [][-1][DotNetBuild]   dotnet C:\Program Files\dotnet\dotnet.exe
VERBOSE: [][-1][DotNetBuild]   dotnet 2.1.4
VERBOSE: [][-1][DotNetBuild]   dotnet --configuration=Debug
VERBOSE: [][-1][DotNetBuild]          -p:Version=1.2.3-rc1
```

Reply:
I want it to show:
* the full path of dotnet.exe and the version the sdk and version of dotnet.nexe being used (as verbose)
* then the actual command it's running, with parameters seperated by spaces and enclosed with quotes if the argument contains spaces (as information)
* the individual parameters on separate lines as verbose messages.

Reply:
I've since added the full command being executed as an info message as per your other comment in this PR. It looks like this:
```
[][-1][DotNetBuild]  & "C:\Program Files\dotnet\dotnet.exe" build --configuration=Debug -p:Version=1.2.3-rc1  C:\Build\CoreProject\DotNetCore.csproj
```

The verbose messages, that's essentially what I previously had:
```
VERBOSE: [][-1][DotNetBuild]  dotnet.exe command path:    C:\Program Files\dotnet\dotnet.exe
VERBOSE: [][-1][DotNetBuild]  dotnet.exe SDK version:     2.1.4
VERBOSE: [][-1][DotNetBuild]  dotnet.exe arguments:       --configuration=Debug
VERBOSE: [][-1][DotNetBuild]                              -p:Version=1.2.3-rc1
```
However you requested I change it to `dotnet {0}`. Were you wanting the description removed from each line entirely so it looks like this?:
```
VERBOSE: [][-1][DotNetBuild]  C:\Program Files\dotnet\dotnet.exe
VERBOSE: [][-1][DotNetBuild]  2.1.4
VERBOSE: [][-1][DotNetBuild]  --configuration=Debug
VERBOSE: [][-1][DotNetBuild]  -p:Version=1.2.3-rc1
```
Trying to figure out what exactly to change about the original verbose messages.

Reply:
Just show the version of dotnet.exe as a verbose message. Remove the others.

Reply:
What is the `$Prefix` parameter meant to do? It's never used in the above function. If defined, is it supposed to prefix every logged message?

Reply:
Remove it. I was thinking this would be used outside tasks, but in this first iteration, it won't be.

Reply:
I had assumed as much and already removed it in the initial commit of the function.